core/arch/arm/kernel/thread_optee_smc.c

```diff
@@ -157,9 +157,7 @@ static uint32_t call_entry_std(struct optee_msg_arg *arg, size_t num_params,
 		 * struct. But if it is we must use the supplied struct and
 		 * at the same time make sure to not break anything.
 		 */
-		if (IS_ENABLED(CFG_PREALLOC_RPC_CACHE) |
-		    thread_prealloc_rpc_cache)
-			clear_prealloc_rpc_cache(thr);
+		clear_prealloc_rpc_cache(thr);
```

Comment:
I'm not so keen on doing this unconditionally. This if-case is our new optimized common path and that shouldn't do more than what's needed.